yi-fuzzy-open/src/Yi/Fuzzy.hs

```diff
@@ -76,133 +89,134 @@ fuzzyOpen = fuzzyOpenWithDepth defaultDepth
 
 -- | Fuzzy-opens the directory to the specified depth. The depth needs
 -- to be at least @1@ for it to do anything meaningful.
-fuzzyOpenWithDepth :: Int -> YiM ()
-fuzzyOpenWithDepth d = case () of
-  _ | d <= 0 -> printMsg "You need at least depth of 1 for fuzzyOpenWithDepth"
-    | otherwise -> do
-    fileList <- fmap (fmap FileItem)
-                     (liftBase (getRecursiveContents d "."))
-    bufList <- fmap (fmap (BufferItem . ident . attributes))
-                    (withEditor (gets (M.elems . buffers)))
-    promptRef <- withEditor (spawnMinibufferE "" (const localKeymap))
-    let initialState =
-            FuzzyState (fileList <> V.fromList bufList)
-                       (Just 0)
-                       ""
-    withGivenBuffer promptRef $ do
-        putBufferDyn initialState
-    withEditor (renderE initialState)
+fuzzyOpenWithDepth :: Natural -> YiM ()
+fuzzyOpenWithDepth d = do
+  fileList  <- (fmap . fmap) (FileItem . T.pack) (liftBase $ getRecursiveContents d ".")
+  bufList   <- (fmap . fmap) (BufferItem . ident . attributes) (withEditor (gets (M.elems . buffers)))
+  promptRef <- withEditor (spawnMinibufferE "" (const localKeymap))
+
+  let initialState = FuzzyState (PL.fromList (filterNotCommon bufList <> fileList)) ""
+  withGivenBuffer promptRef $ putBufferDyn initialState
+  withEditor (renderE initialState)
+ where
+  filterNotCommon :: [FuzzyItem] -> [FuzzyItem]
+  filterNotCommon = filter ((\n -> not (n == "console" || n == "messages")) . itemAsTxt)
+
 
 -- shamelessly stolen from Chapter 9 of Real World Haskell
```

Comment:
I think you can delete that note now that you've pretty much rewritten it.

Reply:
👍